velox/experimental/breeze/CMakeLists.txt

```
@@ -0,0 +1,150 @@
+
+cmake_policy(SET CMP0116 NEW)
+
+# Sets new behavior for CMP0135, which controls how timestamps are extracted
+# when using ExternalProject_Add():
+# https://cmake.org/cmake/help/latest/policy/CMP0135.html
+if(POLICY CMP0135)
+  cmake_policy(SET CMP0135 NEW)
+  set(CMAKE_POLICY_DEFAULT_CMP0135 NEW)
+endif()
+
+project(breeze)
+
+option(BUILD_TESTING "Build testing." ON)
+option(BUILD_CUDA "Build CUDA tests." OFF)
+option(BUILD_HIP "Build HIP tests." OFF)
+option(BUILD_SYCL "Build SYCL tests." OFF)
+option(BUILD_OPENCL "Build OpenCL tests." OFF)
+option(BUILD_OPENMP "Build OpenMP tests." OFF)
+option(BUILD_METAL "Build Metal tests." OFF)
+option(BUILD_TRACING "Build tracing." ON)
+
+if(NOT DEFINED PERFTEST_EXT_TYPES)
+  set(PERFTEST_EXT_TYPES
+      0
+      CACHE STRING "Extended test types for perf tests")
+endif()
+
+if(NOT DEFINED VELOX_PROJECT_SOURCE_DIR)
+  set(VELOX_PROJECT_SOURCE_DIR "${PROJECT_SOURCE_DIR}/../../..")
+endif()
+
+list(
+  PREPEND
+  CMAKE_MODULE_PATH
+  "${PROJECT_SOURCE_DIR}/cmake"
+  "${VELOX_PROJECT_SOURCE_DIR}/CMake"
+  "${VELOX_PROJECT_SOURCE_DIR}/CMake/third-party")
+
+# Include Velox ThirdPartyToolchain dependencies macros
+include(ResolveDependency)
+
+set(OPT_FLAGS "-O0;-g")
+set(BREEZE_BUILD_TYPE "${CMAKE_BUILD_TYPE}")
+if(CMAKE_BUILD_TYPE MATCHES Release)
+  set(NDEBUG_DEFINE "-DNDEBUG")
+  set(OPT_FLAGS "-O2")
+  set(LIB_OPT_FLAGS "-O2")
+endif()
+if(CMAKE_BUILD_TYPE MATCHES RelWithDebInfo)
+  set(OPT_FLAGS "-O2;-g")
+  # override build type for debug info to be limited to code built with
+  # OPT_FLAGS
+  set(CMAKE_BUILD_TYPE "Release")
+endif()
+if(CMAKE_BUILD_TYPE MATCHES Asan)
+  set(OPT_FLAGS "-O1;-g")
+  set(SANITIZE_COMPILE_FLAGS "-fsanitize=address;-fno-omit-frame-pointer")
+  set(SANITIZE_LINK_FLAGS "-fsanitize=address;-fno-omit-frame-pointer")
+endif()
+
+set(WARN_FLAGS "-Wall;-Wextra;-Werror")
+set(COMPILER_WARN_FLAGS "-Xcompiler=,-Wall,-Wextra,-Werror")
+
+if(BUILD_CUDA)
+  include(cuda)
+endif()
+
+if(BUILD_HIP)
+  include(hip)
+endif()
+
+if(BUILD_SYCL)
+  include(sycl)
+endif()
+
+if(BUILD_OPENCL)
+  include(opencl)
+endif()
+
+if(BUILD_OPENMP)
+  include(openmp)
+endif()
+
+if(BUILD_METAL)
+  include(metal)
+endif()
+
+if(BUILD_TRACING)
+  find_package(Threads REQUIRED)
+
+  # static library for perfetto
+  include_directories(external/perfetto/sdk)
+  add_library(perfetto STATIC ../../external/perfetto/sdk/perfetto.cc)
+  target_compile_features(perfetto PRIVATE cxx_std_17)
+  target_compile_options(perfetto PRIVATE -fPIC)
+
+  set(TRACING_LIB "perfetto")
+  set(TRACING_LIBS "$<TARGET_FILE_DIR:perfetto>/lib${TRACING_LIB}.a")
+  set(TRACING_INCLUDE "-I${CMAKE_SOURCE_DIR}/../../external/perfetto/sdk")
+  set(TRACING_DEFINE "TRACING=1")
+  set(HOST_TRACING_FLAGS "-DHOST_TRACING=1;${TRACING_INCLUDE}")
+endif()
+
+if(NOT DISABLE_GOOGLETEST)
+  set(gtest_SOURCE AUTO)
+  resolve_dependency(gtest)
+endif()
+
+if(BUILD_TESTING)
+  set(BUILD_TESTING ON)
+  enable_testing()
+  include(CTest)
```

```suggestion
if(BUILD_TESTING)
  set(BUILD_TESTING ON)
  enable_testing()
  include(CTest)
```
```suggestion
if(BUILD_TESTING)
  include(CTest)
```
Including CTest enables testing, except if the option is off (which would also skip the if), so you could even included it outside of the if but it makes sense this way :)

---

Yep. Fixed in latest version.